app/src/main/java/org/mozilla/fenix/utils/Undo.kt

```diff
@@ -45,7 +43,7 @@ fun allowUndo(
         .setText(message)
         .setAction(undoActionTitle) {
             requestedUndo.set(true)
-            mainScope.launch {
+            launch {
```

Comment:
Should we have withContext(Main)? UI operations like snackbar.dismiss() cannot be performed on background threads.

Reply:
Should we allow specify dispatcher for the `operation.invoke()` as well?